### PR TITLE
[WIP] Fix issue 166

### PR DESCRIPTION
**Pull request by Thomasb81**
Hello 

Current draft push request  is an attempt to fix : https://github.com/Nic30/hdlConvertor/issues/166

First it remove new line added in preprocessing  result.  One part of the fix consist to make the preprocessing grammar to consume the training new line. The Ieee1800-2017  chapter 22.4 says
> Only white space or a comment may appear on the same line as the <code>`include</code> compiler directive.

So we can remove new line on the line containing directive.  
If included file does not end by new line, next line will appear on the last line from the included file. See tests/sv_pp/expected/include_many_dir/from_subdirectory.txt

Second part consist to enrich token build by antlr4 with original file location using ` file_line_map` data.
But the result is not yet correct: current data structure does not allow to retrieve all information we need for every lines.

The CodePosition object has been update to reflect the ` fileName`  data. hdlConvertorAst would also need to be updated.

### CODE REVIEW

**Comment by Thomasb81** [comment]
Tests failing because hdlConvertorAst requiere an update of ` Codeposition`  object.

**Comment by Nic30** [comment]
@Thomasb81
Can you summarize the state of this issue, optionally can you give me info about what is need to be done?

**Comment by Thomasb81** [comment]
Also here https://github.com/Nic30/hdlConvertor/issues/134#issuecomment-681099582 you mention to use smart_prt. to reference fileName in codePosition. Current code make a copy.
